test/integration/test_connection_management.py

```diff
@@ -38,6 +38,10 @@ def _run_with_number_of_connections(fixture,
   return counters
 
 
+def _count_log_lines_with_substring(logs, substring):
```

Comment:
This likely deserves to be a function in a separate library with its own unit test.

We could add it to one of the existing helper libraries, or define a new one. The existing unit tests are here:

https://github.com/envoyproxy/nighthawk/tree/main/test/integration/unit_tests

Reply:
Done.